pkg/reconciler/v1alpha1/certificate/resources/cert_manager_certificate_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	certmanagerv1alpha1 "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"github.com/knative/pkg/kmeta"
+	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"github.com/knative/serving/pkg/reconciler/v1alpha1/certificate/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var cert = &v1alpha1.Certificate{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "test-cert",
+		Namespace: "test-ns",
+	},
+	Spec: v1alpha1.CertificateSpec{
+		DNSNames:   []string{"host1.example.com", "host2.example.com"},
+		SecretName: "secret0",
+	},
+}
+
+var cmConfig = &config.CertManagerConfig{
+	SolverConfig: &certmanagerv1alpha1.SolverConfig{
+		DNS01: &certmanagerv1alpha1.DNS01SolverConfig{
+			Provider: "cloud-dns-provider",
+		},
+	},
+	IssuerRef: &certmanagerv1alpha1.ObjectReference{
+		Kind: "ClusterIssuer",
+		Name: "Letsencrypt-issuer",
+	},
+}
+
+func TestMakeCertManagerCertificate(t *testing.T) {
+	tests := []struct {
+		name     string
+		knCert   *v1alpha1.Certificate
+		cmConfig *config.CertManagerConfig
+		want     *certmanagerv1alpha1.Certificate
+	}{{
```

Comment:
you seem to have a single test. Why do you need the loop?

Reply:
Done